File: BackEnd/Application/UseCases/TicketStatuses/UpdateTicketStatus/UpdateTicketStatusCommandHandler.cs
using Domain.Aggregates;
using Infrastructure.Repository;
using Infrastructure.UnitOfWork;
using MediatR;

namespace Application.UseCases.TicketStatuses.UpdateTicketStatus;

public class UpdateTicketStatusCommandHandler(IRepository<TicketStatus, Guid> ticketStatusRepository, IUnitOfWork unitOfWork) : IRequestHandler<UpdateTicketStatusCommand, bool>
{
    public async Task<bool> Handle(UpdateTicketStatusCommand request, CancellationToken cancellationToken)
    {
        var ticketStatus = await ticketStatusRepository.GetByIdAsync(request.Id, cancellationToken);
        if (ticketStatus == null) return false;
        ticketStatus.Update(request.Name);
        await ticketStatusRepository.UpdateAsync(ticketStatus);
        await unitOfWork.SaveChangesAsync(cancellationToken);
        return true;
    }
}

File: flutter/lib/application/use_cases/ticket_statuses/update_ticket_status/update_ticket_status_command_handler.dart
import 'package:enterprise_management/application/use_cases/ticket_statuses/update_ticket_status/update_ticket_status_command.dart';
import 'package:enterprise_management/infrastructure/repositories/ticket_status_repository.dart';
import 'package:enterprise_management/shared_kernel/cqrs/command_handler.dart';
import 'package:enterprise_management/shared_kernel/result/result.dart';

class UpdateTicketStatusCommandHandler extends ICommandHandler<UpdateTicketStatusCommand, bool> {
  const UpdateTicketStatusCommandHandler({required this._ticketStatusRepository});
  final ITicketStatusRepository _ticketStatusRepository;

  @override
  Future<Result<bool>> handle(UpdateTicketStatusCommand command) async {
    final res = await _ticketStatusRepository.updateTicketStatus(id: command.id, name: command.name, projectIds: command.projectIds);
    return Result.success(res);
  }

}